pkg/cluster/config.go

```diff
@@ -107,23 +109,29 @@ func NewConfig(restConfig *rest.Config) (Config, error) {
 	}, nil
 }
 
-// validateK8sVersion checks for valid k8s version in the cluster. It returns an error for all versions not equal
-// to supported major version. This is being done this way, and not by directly getting the cluster version, as OpenShift CI
-// returns version in the format 0.0.x and not the actual version attached to its clusters.
+// validateK8sVersion checks for valid k8s version in the cluster. It returns an error for all versions that are not in
+// range of given base version(x.y.z) and x.y+1.z version.
 func (c *config) validateK8sVersion() error {
 	versionInfo, err := c.oclient.Discovery().ServerVersion()
 	if err != nil {
 		return errors.Wrap(err, "error retrieving server version ")
 	}
-	// split the version in the form Major.Minor. For e.g v1.18.0-rc.1 -> 1.18
-	k8sVersion := strings.TrimLeft(versionInfo.GitVersion, "v")
-	clusterBaseVersion := strings.Join(strings.SplitN(k8sVersion, ".", 3)[:2], ".")
+	// split the version in the form Major.Minor. For e.g v1.18.0-rc.1 -> v1.18
+	clusterBaseVersion := semver.MajorMinor(versionInfo.GitVersion)
+	// Convert base version to float and add 1 to Minor version
+	baseVersion, err := strconv.ParseFloat(strings.TrimPrefix(baseK8sVersion, "v"), 64)
+	if err != nil {
+		return errors.Wrap(err, "error converting kube MajorMinor version to float")
```

Comment:
`major.minor`. Moreover it will help if you mention the `baseK8sVersion` in the error. i.e. `error converting k8s version 1.20 to float`.